src/main/scala/epic/logo/Inferencer.scala

```diff
@@ -0,0 +1,7 @@
+package epic.logo
+
+trait Inferencer[S] {
+  def initialState: S
+  def reduceStates(state1: S, state2: S): S
```

Comment:
yes please

On Thu, May 26, 2016 at 8:26 PM, adampauls <notifications@github.com> wrote:

> In src/main/scala/epic/logo/Inferencer.scala
> <https://github.com/dlwh/epic/pull/53#discussion_r64852990>:
>
> > @@ -0,0 +1,7 @@
> > +package epic.logo
> > +
> > +trait Inferencer[S] {
> > +  def initialState: S
> > +  def reduceStates(state1: S, state2: S): S
>
> Arbitrary state that the implementers of inferencers may want to carry
> around. The abstraction is like mapreduce: you return some stuff, and also
> say how to combine that stuff, then you get the combined stuff in various
> callbacks. Allows inferencers to have internal state without worrying about
> threadsafety. Sounds like I need some comments.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dlwh/epic/pull/53/files/bdf0547a0919d604cf236bdc542181e413f7e9a8#r64852990>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAloTkt927OW3um6hasHga5VroxisuLks5qFmR7gaJpZM4In_F1>
> .
>